zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -434,7 +434,12 @@ protected Object jobRun() throws Throwable {
     try {
       InterpreterContext context = getInterpreterContext();
       InterpreterContext.set(context);
-      InterpreterResult ret = interpreter.interpret(script, context);
+      UserCredentials creds = context.getAuthenticationInfo().getUserCredentials();
```

Comment:
Could we introduce interpreter local property (via `context.getLocalProperties`) that controls whether to apply injecting password ? Because this approach may introduce 2 issues:
* performance impact, especially when the code or interpreter result is large. 
* code or interpreter result may be corrupted.

Here's the local property sample
```
%sql(injectCredentials=true)
select ...

```

Reply:
I will work on adding that in. Would it make sense to have the property be an interpreter config setting as well? I am hoping to get this PR into both 0.8.X and 0.9 series, and currently 0.8.X does not support local properties, but I think it makes sense to have credential injection be opt-in.

Reply:
As interpreter setting makes sense for me. Local property will take precedence over interpreter setting.